### ISSUE TITLE
Improve DAP configurations for C/C++

### ISSUE THREAD

**Issue by peter-bread** [peter-bread/peter.nvim#25]
Add more DAP configs, e.g. using build tools like make and task so i dont need to leave nvim to compile before debugging.

**Comment by peter-bread** [comment]
#378b51ba60788dcab50a6a9d13f8c7848f97cf94

Find `Makefile` and execute `make` with no arguments before starting debug session.

**Comment by peter-bread** [comment]
Now I've added overseer, this should improve this. Still need to properly configure overseer and integrate it with dap.

**Comment by peter-bread** [comment]
On hold as I have other priorities right now.
